controllers/volsync/vshandler.go

```diff
@@ -931,21 +919,37 @@ func (v *VSHandler) validateSnapshotAndEnsurePVC(rdSpec ramendrv1alpha1.VolSyncR
 		return err
 	}
 
-	var pvc *corev1.PersistentVolumeClaim
+	if v.IsCopyMethodDirect() {
+		// Directly use the RD pvc
+		v.log.V(1).Info("Using copyMethod Direct", "latestImage", snapshotRef, "pvcName", rdSpec.ProtectedPVC.Name)
 
-	var restoreSize *resource.Quantity
+		pvc := &corev1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      rdSpec.ProtectedPVC.Name,
+				Namespace: v.owner.GetNamespace(),
+			},
+		}
 
-	if snap.Status != nil {
-		restoreSize = snap.Status.RestoreSize
-	}
+		err = ValidateObjectExists(v.ctx, v.client, pvc)
+		if err != nil {
+			return err
+		}
+	} else {
+		// Restore pvc from snapshot
+		var restoreSize *resource.Quantity
 
-	pvc, err = v.ensurePVCFromSnapshot(rdSpec, snapshotRef, restoreSize)
-	if err != nil {
-		return err
+		if snap.Status != nil {
+			restoreSize = snap.Status.RestoreSize
+		}
+
+		_, err := v.ensurePVCFromSnapshot(rdSpec, snapshotRef, restoreSize)
+		if err != nil {
+			return err
+		}
 	}
 
 	// Add ownerRef on snapshot pointing to the pvc - if/when the PVC gets cleaned up, then GC can cleanup the snap
```

Comment:
Do you mind updating this comment?  Just so it's clear that we're making the vrg the owner of the snapshot now instead of the pvc

Reply:
Updated